src/frontend/src/optimizer/plan_node/stream_local_simple_agg.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+
+use itertools::Itertools;
+use risingwave_common::catalog::{DatabaseId, SchemaId};
+use risingwave_pb::stream_plan::stream_node::NodeBody as ProstStreamNode;
+
+use super::logical_agg::PlanAggCall;
+use super::{LogicalAgg, PlanBase, PlanRef, PlanTreeNodeUnary, ToStreamProst};
+use crate::optimizer::property::RequiredDist;
+
+#[derive(Debug, Clone)]
+pub struct StreamLocalSimpleAgg {
+    pub base: PlanBase,
+    logical: LogicalAgg,
+}
+
+impl StreamLocalSimpleAgg {
+    pub fn new(logical: LogicalAgg) -> Self {
+        let ctx = logical.base.ctx.clone();
+        let pk_indices = logical.base.pk_indices.to_vec();
+        let input = logical.input();
+        let input_dist = input.distribution();
+        debug_assert!(input_dist.satisfies(&RequiredDist::AnyShard));
+
+        let append_only = input.append_only();
```

Comment:
`AggCall` uses plan node's `append_only` field: https://github.com/singularity-data/risingwave/blob/91a699062b6ea2d20f4f81b485f8334b283b36d8/src/stream/src/from_proto/local_simple_agg.rs#L37-L53.

`append_only` of GlobalAgg depends on its inputs. Which can be traced back to `stream_local_simple_agg`.

We need `AggCall` to be `append_only` to use `SingleValue` state rather than extreme value:

https://github.com/singularity-data/risingwave/blob/91a699062b6ea2d20f4f81b485f8334b283b36d8/src/stream/src/executor/managed_state/aggregation/mod.rs#L129-L134